src/main/docs/guide/httpServer/customHttpStatusCode.adoc

```diff
@@ -0,0 +1,39 @@
+The Micronaut framework supports custom HTTP status codes. This feature enables developers to define and use HTTP status codes beyond the standard ones, allowing integration with systems or APIs that rely on non-standard status codes.
+
+=== Custom HTTP Status Codes
+Micronaut supports arbitrary HTTP status codes that are not part of the predefined `HttpStatus` enum. Developers can specify any valid HTTP status code within the range `100-599`.
+
+==== Usage
+- `HttpResponse.status(int statusCode)`:
+Allows setting a custom HTTP status code.
+- `HttpResponse.status(int statusCode, String reason)`:
+Allows setting a custom HTTP status code along with a reason phrase.
+
+Below we can see how we support custom status codes and how to implement them. 
+
+snippet::io.micronaut.http.HttpResponse[tags="getStatusMethod", indent=0, title="HttpResponse.java"]
+Let's start with a simple example. Given the following class:
+
+snippet::io.micronaut.http.client.netty.FullNettyClientHttpResponse[tags="getBodyMethod", indent=0, title="FullNettyClientHttpResponse.java"]
```

Comment:
Snippet is for code samples in test-suite, test-suite-groovy and test-suite-kotlin. This does not work. Please, try `./gradlew docs` locally to see if it renders correctly.